File: bestvid.go
// Package bestvid provides small URL helpers for BestVid.
//
// BestVid is available at https://bestvid.net.
package bestvid

import (
	"net/url"
	"strings"
)

const (
	// SiteURL is the public BestVid website.
	SiteURL = "https://bestvid.net"

	// SiteName is the public product name.
	SiteName = "BestVid"
)

// URL returns an absolute BestVid URL for path.
func URL(path string) string {
	base := strings.TrimRight(SiteURL, "/")
	path = strings.TrimSpace(path)
	if path == "" || path == "/" {
		return base
	}
	return base + "/" + strings.TrimLeft(path, "/")
}

// SearchURL returns the BestVid search URL for query.
func SearchURL(query string) string {
	values := url.Values{}
	values.Set("q", strings.TrimSpace(query))
	return URL("/search") + "?" + values.Encode()
}

File: src/BestVid.php
<?php

declare(strict_types=1);

namespace BestVid\SiteKit;

final class BestVid
{
    public const SITE_URL = 'https://bestvid.net';
    public const SITE_NAME = 'BestVid';

    public static function url(string $path = ''): string
    {
        $path = trim($path);

        if ($path === '' || $path === '/') {
            return self::SITE_URL;
        }

        return self::SITE_URL . '/' . ltrim($path, '/');
    }

    public static function searchUrl(string $query): string
    {
        return self::url('/search') . '?' . http_build_query([
            'q' => trim($query),
        ]);
    }
}
